File: solutions/2026-01-17/cpp/2026-01-17-find-the-largest-area-of-square-inside-two-rectangles.cpp
```cpp
// Problem: Find the Largest Area of Square Inside Two Rectangles
// Link: https://leetcode.com/problems/find-the-largest-area-of-square-inside-two-rectangles/
// Approach:
// The problem asks for the largest square that can fit within the intersection of *any* two given rectangles.
// To find the intersection of two rectangles, we need to find the overlapping region's bottom-left and top-right coordinates.
// For two rectangles defined by (x1_bl, y1_bl), (x1_tr, y1_tr) and (x2_bl, y2_bl), (x2_tr, y2_tr):
// The intersection's bottom-left x-coordinate will be the maximum of the two bottom-left x-coordinates.
// The intersection's bottom-left y-coordinate will be the maximum of the two bottom-left y-coordinates.
// The intersection's top-right x-coordinate will be the minimum of the two top-right x-coordinates.
// The intersection's top-right y-coordinate will be the minimum of the two top-right y-coordinates.
// If the calculated top-right x is less than or equal to the calculated bottom-left x, or the calculated top-right y is less than or equal to the calculated bottom-left y, then the rectangles do not intersect (or intersect at a line/point, which cannot contain a square with positive area).
// If they do intersect, the width of the intersection is `intersect_tr_x - intersect_bl_x` and the height is `intersect_tr_y - intersect_bl_y`.
// The largest square that can fit inside this intersection will have a side length equal to the minimum of its width and height.
// We need to iterate through all possible pairs of rectangles, calculate the intersection for each pair, determine the maximum square side length within that intersection, and keep track of the overall maximum side length found.
// The final answer is the square of this maximum side length.
//
// Time Complexity: O(N^2), where N is the number of rectangles. We iterate through all pairs of rectangles.
// Space Complexity: O(1), as we only use a few variables to store intermediate calculations.
#include <vector>
#include <algorithm>

class Solution {
public:
    long long largestSquareArea(std::vector<std::vector<int>>& bottomLeft, std::vector<std::vector<int>>& topRight) {
        long long max_side = 0; // Stores the maximum side length of a square found so far.
        int n = bottomLeft.size(); // Number of rectangles.

        // Iterate through all possible pairs of rectangles.
        for (int i = 0; i < n; ++i) {
            for (int j = i + 1; j < n; ++j) {
                // Extract coordinates for the first rectangle.
                int bl1_x = bottomLeft[i][0];
                int bl1_y = bottomLeft[i][1];
                int tr1_x = topRight[i][0];
                int tr1_y = topRight[i][1];

                // Extract coordinates for the second rectangle.
                int bl2_x = bottomLeft[j][0];
                int bl2_y = bottomLeft[j][1];
                int tr2_x = topRight[j][0];
                int tr2_y = topRight[j][1];

                // Calculate the coordinates of the intersecting rectangle.
                // The bottom-left x of the intersection is the maximum of the two bottom-left x's.
                int intersect_bl_x = std::max(bl1_x, bl2_x);
                // The bottom-left y of the intersection is the maximum of the two bottom-left y's.
                int intersect_bl_y = std::max(bl1_y, bl2_y);
                // The top-right x of the intersection is the minimum of the two top-right x's.
                int intersect_tr_x = std::min(tr1_x, tr2_x);
                // The top-right y of the intersection is the minimum of the two top-right y's.
                int intersect_tr_y = std::min(tr1_y, tr2_y);

                // Check if the rectangles actually intersect to form a region with positive area.
                // If the calculated top-right x is less than or equal to the bottom-left x, or
                // if the calculated top-right y is less than or equal to the bottom-left y,
                // then there is no valid intersection region for a square.
                if (intersect_tr_x > intersect_bl_x && intersect_tr_y > intersect_bl_y) {
                    // Calculate the width and height of the intersecting region.
                    long long width = intersect_tr_x - intersect_bl_x;
                    long long height = intersect_tr_y - intersect_bl_y;

                    // The largest square that can fit inside this intersection
                    // has a side length equal to the minimum of its width and height.
                    long long current_side = std::min(width, height);

                    // Update the overall maximum side length if the current one is larger.
                    max_side = std::max(max_side, current_side);
                }
            }
        }

        // The problem asks for the area, which is the square of the maximum side length.
        return max_side * max_side;
    }
};
```

File: solutions/2026-01-17/java/2026-01-17-find-the-largest-area-of-square-inside-two-rectangles.java
```java
/**
 * Problem: Find the Largest Area of Square Inside Two Rectangles
 * Link: https://leetcode.com/problems/find-the-largest-area-of-square-inside-two-rectangles/
 *
 * Approach:
 * The problem asks for the maximum area of a square that can fit inside the intersection of any two given rectangles.
 * To find the intersection of two rectangles, we need to determine the overlapping region.
 * Given two rectangles defined by their bottom-left (x1, y1) and top-right (x2, y2) coordinates:
 * Rectangle 1: (x1_1, y1_1) to (x2_1, y2_1)
 * Rectangle 2: (x1_2, y1_2) to (x2_2, y2_2)
 *
 * The intersection's bottom-left corner will have coordinates:
 * intersect_x1 = max(x1_1, x1_2)
 * intersect_y1 = max(y1_1, y1_2)
 *
 * The intersection's top-right corner will have coordinates:
 * intersect_x2 = min(x2_1, x2_2)
 * intersect_y2 = min(y2_1, y2_2)
 *
 * For an intersection to exist, we must have intersect_x1 < intersect_x2 and intersect_y1 < intersect_y2.
 * If an intersection exists, its width is `intersect_x2 - intersect_x1` and its height is `intersect_y2 - intersect_y1`.
 * The largest square that can fit inside this intersection will have a side length equal to the minimum of its width and height:
 * side_length = min(intersect_x2 - intersect_x1, intersect_y2 - intersect_y1).
 * The area of this square is side_length * side_length.
 *
 * We need to iterate through all possible pairs of rectangles, calculate the area of the largest square that can fit in their intersection,
 * and keep track of the maximum area found.
 *
 * The algorithm proceeds as follows:
 * 1. Initialize `maxArea` to 0.
 * 2. Iterate through all distinct pairs of rectangles (i, j) where i < j.
 * 3. For each pair, extract the coordinates of rectangle i: (bl_x1, bl_y1) and (tr_x1, tr_y1).
 * 4. Extract the coordinates of rectangle j: (bl_x2, bl_y2) and (tr_x2, tr_y2).
 * 5. Calculate the coordinates of the intersection:
 *    `intersect_bl_x = Math.max(bl_x1, bl_x2)`
 *    `intersect_bl_y = Math.max(bl_y1, bl_y2)`
 *    `intersect_tr_x = Math.min(tr_x1, tr_x2)`
 *    `intersect_tr_y = Math.min(tr_y1, tr_y2)`
 * 6. Check if the intersection is valid: `intersect_bl_x < intersect_tr_x` and `intersect_bl_y < intersect_tr_y`.
 * 7. If the intersection is valid:
 *    a. Calculate the width: `width = intersect_tr_x - intersect_bl_x`
 *    b. Calculate the height: `height = intersect_tr_y - intersect_bl_y`
 *    c. Calculate the side length of the largest square: `side = Math.min(width, height)`
 *    d. Calculate the area of this square: `currentArea = side * side`
 *    e. Update `maxArea = Math.max(maxArea, currentArea)`
 * 8. After checking all pairs, return `maxArea`.
 *
 * Time Complexity:
 * We iterate through all pairs of n rectangles. The number of pairs is n * (n-1) / 2, which is O(n^2).
 * For each pair, the intersection calculation and area calculation take constant time, O(1).
 * Therefore, the overall time complexity is O(n^2).
 *
 * Space Complexity:
 * We are only using a few variables to store coordinates and the maximum area. The space used does not depend on the input size n.
 * Therefore, the space complexity is O(1).
 */
class Solution {
    /**
     * Calculates the maximum area of a square that can fit inside the intersecting region of at least two rectangles.
     *
     * @param bottomLeft A 2D array where bottomLeft[i] = [x, y] represents the bottom-left corner of the ith rectangle.
     * @param topRight A 2D array where topRight[i] = [x, y] represents the top-right corner of the ith rectangle.
     * @return The maximum area of a square that can fit inside the intersection of any two rectangles, or 0 if no such square exists.
     */
    public long largestSquareArea(int[][] bottomLeft, int[][] topRight) {
        // Initialize the maximum area found so far to 0.
        long maxArea = 0;
        // Get the number of rectangles.
        int n = bottomLeft.length;

        // Iterate through all distinct pairs of rectangles.
        // We use nested loops where the outer loop goes from i=0 to n-2,
        // and the inner loop goes from j=i+1 to n-1. This ensures each pair is considered exactly once.
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {

                // Extract coordinates for the first rectangle (rectangle i).
                int bl_x1 = bottomLeft[i][0];
                int bl_y1 = bottomLeft[i][1];
                int tr_x1 = topRight[i][0];
                int tr_y1 = topRight[i][1];

                // Extract coordinates for the second rectangle (rectangle j).
                int bl_x2 = bottomLeft[j][0];
                int bl_y2 = bottomLeft[j][1];
                int tr_x2 = topRight[j][0];
                int tr_y2 = topRight[j][1];

                // Calculate the bottom-left corner of the intersection.
                // The x-coordinate is the maximum of the two bottom-left x-coordinates.
                // The y-coordinate is the maximum of the two bottom-left y-coordinates.
                int intersect_bl_x = Math.max(bl_x1, bl_x2);
                int intersect_bl_y = Math.max(bl_y1, bl_y2);

                // Calculate the top-right corner of the intersection.
                // The x-coordinate is the minimum of the two top-right x-coordinates.
                // The y-coordinate is the minimum of the two top-right y-coordinates.
                int intersect_tr_x = Math.min(tr_x1, tr_x2);
                int intersect_tr_y = Math.min(tr_y1, tr_y2);

                // Check if the intersection is valid.
                // A valid intersection exists only if the bottom-left corner is strictly to the left
                // of the top-right corner (intersect_bl_x < intersect_tr_x) AND
                // the bottom-left corner is strictly below the top-right corner (intersect_bl_y < intersect_tr_y).
                if (intersect_bl_x < intersect_tr_x && intersect_bl_y < intersect_tr_y) {
                    // If the intersection is valid, calculate its width and height.
                    int width = intersect_tr_x - intersect_bl_x;
                    int height = intersect_tr_y - intersect_bl_y;

                    // The side length of the largest square that can fit inside this intersection
                    // is the minimum of its width and height.
                    int side = Math.min(width, height);

                    // Calculate the area of this square.
                    // We use 'long' for area to prevent potential integer overflow, as side length can be up to 10^7.
                    long currentArea = (long) side * side;

                    // Update the maximum area found so far if the current square's area is larger.
                    maxArea = Math.max(maxArea, currentArea);
                }
            }
        }

        // Return the overall maximum area found among all intersecting pairs.
        return maxArea;
    }
}
```